frappe_telegram/frappe_telegram/doctype/telegram_bot/telegram_bot.py

```diff
@@ -12,6 +13,23 @@ def autoname(self):
     def validate(self):
         self.set_nginx_path()
 
+    def after_insert(self):
+        default_bot = frappe.db.get_default(DEFAULT_TELEGRAM_BOT_KEY)
+        if not default_bot:
+            self.mark_as_default()
+
+    def after_delete(self):
+        default_bot = frappe.db.get_default(DEFAULT_TELEGRAM_BOT_KEY)
+
+        if default_bot == self.name:
```

Comment:
Uhh, this will cause issues, as the default will point to a deleted bot if all bots are deleted. Integrity issues. 😅

Reply:
The issues are:
1. When the last bot is deleted, the default will not be changed, so it will lead to the default being a bot that is already deleted. The "default bot" being a bot that's already deleted is not good, data integrity-wise.
2. Then after a new bot is created, since the default was not set to null, the new bot will not be marked as the default bot. In fact, the default bot will get stuck as is, unless someone manually marks a new one as default.

All this not to say that the above cannot be implemented, but I will have to change how the default is updated. Instead of checking the default value, we can check the number of current existing bots and update based on that.

Reply:
@Abadulrehman true 👍🏼 i think next step for abstraction will be to make a set of utility functions - which is overkill 😅 let's leave it here 👍🏼